connectors/sources/confluence.py

```diff
@@ -588,20 +854,43 @@ async def _attachment_coro(self, document):
 
     async def _space_coro(self):
         """Coroutine to add spaces documents to Queue"""
-        async for space in self.fetch_spaces():
+        async for space, permissions in self.fetch_spaces():
+            access_control = list(
+                self._get_access_control_from_permission(
+                    permissions=permissions, target_type=SPACE
+                )
+            )
+            space = self._decorate_with_access_control(
+                document=space, access_control=access_control
+            )
             await self.queue.put((space, None))  # pyright: ignore
         await self.queue.put(END_SIGNAL)  # pyright: ignore
 
-    async def _page_blog_coro(self, api_query):
+    async def _page_blog_coro(self, api_query, target_type):
         """Coroutine to add pages/blogposts to Queue
 
         Args:
             api_query (str): API Query Parameters for fetching page/blogpost
+            target_type (str): Type of object to filter permission
         """
-        async for document, attachment_count in self.fetch_documents(api_query):
+        async for document, attachment_count, permissions, restrictions in self.fetch_documents(
```

Comment:
The return type of `fetch_documents` seems to be pretty complex. Can we simplify it? Also what's the difference between `permissions` and `restrictions`?

Reply:
Permission are related to space and Restriction are related to pages & blogpost (Implement side need to override space permission by pages & blogpost restriction). [Difference Reference](https://support.atlassian.com/confluence-cloud/docs/what-are-confluence-cloud-permissions-and-restrictions/).

Reply:
Could we add some comment explaining this difference in the code?

Reply:
Sure!